src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/EditProceedingsAndCostsSectionController.java

```diff
@@ -755,4 +822,379 @@ public String proceedingsActionConfirmPost(
+  }
+
+  /**
+   * Displays the confirmation page for scope limitations with the currently
+   * selected scope limitation and its details.
+   *
+   * @param scopeLimitation the current scope limitation, retrieved from the session.
+   * @param scopeLimitationFlow the scope limitation flow data, retrieved from the session.
+   * @param model the {@link Model} object for passing attributes to the view.
+   * @return the name of the view to render.
+   */
+
+  @GetMapping("/application/proceedings/scope-limitations/confirm")
+  public String scopeLimitationConfirm(
+      @SessionAttribute(CURRENT_SCOPE_LIMITATION) final ScopeLimitation scopeLimitation,
+      @SessionAttribute(SCOPE_LIMITATION_FLOW_FORM_DATA)
+      final ScopeLimitationFlowFormData scopeLimitationFlow,
+      final Model model) {
+
+    model.addAttribute(CURRENT_SCOPE_LIMITATION, scopeLimitation);
+    model.addAttribute(SCOPE_LIMITATION_FLOW_FORM_DATA, scopeLimitationFlow);
+
+    return "application/proceedings-scope-limitations-confirm";
+  }
+
+  /**
+   * Processes the confirmation of scope limitations, adding or updating them
+   * based on the action specified in the proceeding flow. Handles both new additions
+   * and updates to existing scope limitations within a session or proceeding.
+   *
+   * @param scopeLimitation the scope limitation to be confirmed, retrieved from the session.
+   * @param proceedingFlow the proceeding flow data, indicating the current action.
+   * @param scopeLimitationFlow the scope limitation flow data, containing index information
+   *                            for updates.
+   * @param user the current user's details, for updating proceedings.
+   * @param model the {@link Model} object for passing attributes to the view.
+   * @param session the {@link HttpSession} object for accessing session attributes.
+   * @return the redirect URL for the proceeding confirmation page.
+   */
+  @PostMapping("/application/proceedings/scope-limitations/confirm")
+  public String scopeLimitationConfirmPost(
+      @SessionAttribute(CURRENT_SCOPE_LIMITATION) final ScopeLimitation scopeLimitation,
+      @SessionAttribute(PROCEEDING_FLOW_FORM_DATA) final ProceedingFlowFormData proceedingFlow,
+      @SessionAttribute(SCOPE_LIMITATION_FLOW_FORM_DATA)
+      final ScopeLimitationFlowFormData scopeLimitationFlow,
+      @SessionAttribute(USER_DETAILS) final UserDetail user,
+      final Model model,
+      final HttpSession session) {
+
+
+    if (proceedingFlow.getAction().equals(ACTION_ADD)) {
+      final List<ScopeLimitation> scopeLimitations =
+          (List<ScopeLimitation>) session.getAttribute(PROCEEDING_SCOPE_LIMITATIONS);
+
+      model.addAttribute(PROCEEDING_FLOW_FORM_DATA, proceedingFlow);
+
+      if (scopeLimitationFlow.getScopeLimitationIndex() == null) {
+        scopeLimitations.add(scopeLimitation);
+      } else {
+        scopeLimitations.set(scopeLimitationFlow.getScopeLimitationIndex(), scopeLimitation);
```

Comment:
Should probably check the index is within bounds

Reply:
amended, added a throw condition